vue-app/src/components/RecipientSubmissionWidget.vue

```diff
@@ -165,67 +169,120 @@ export default class RecipientSubmissionWidget extends Vue {
+  }
+
   get depositToken(): string {
-    return this.registryInfo?.depositToken ?? ''
+    return this.$store.state.recipientRegistryInfo?.depositToken ?? ''
   }
 
   handleSubmit(): void {
     this.addRecipient()
   }
 
+  public calculateFiatFee(ethAmount: BigNumber): string {
+    if (this.$store.state.recipientRegistryInfo && this.ethPrice) {
+      return Number(
+        this.ethPrice.ethereum.usd * Number(formatAmount(ethAmount, 18))
+      ).toFixed(2)
+    }
+    return '-'
+  }
+
+  private async getEstimatedGasFee(): Promise<BigNumber> {
+    const { recipient, recipientRegistryAddress, recipientRegistryInfo } =
+      this.$store.state
+
+    if (
+      recipientRegistryInfo &&
+      this.ethPrice &&
+      this.walletProvider &&
+      recipientRegistryAddress &&
+      recipient
+    ) {
+      const registry = new Contract(
+        recipientRegistryAddress,
+        OptimisticRecipientRegistry,
+        this.walletProvider
+      )
+
+      return await registry.estimateGas.addRecipient(
+        recipient.fund.address,
+        JSON.stringify({
+          ...recipient.fund,
+          ...recipient.links,
+          ...recipient.image,
+          ...recipient.project,
+        }),
+        { value: recipientRegistryInfo.deposit }
+      )
+    }
+    return BigNumber.from(0)
+  }
+
   private async addRecipient() {
-    // TODO move to create()?
-    const recipientRegistryAddress = this.$store.state.recipientRegistryAddress
-    const signer = this.$store.state.currentUser.walletProvider.getSigner()
-    const recipient = this.$store.state.recipient
-    // TODO use this.registryInfo
-    const registryInfo = await getRegistryInfo(
-      this.$store.state.recipientRegistryAddress
-    )
+    const {
+      currentUser,
+      recipient,
+      recipientRegistryAddress,
+      recipientRegistryInfo,
+    } = this.$store.state
 
     // TODO where to make `isPending` vs. `isWaiting`? In `waitForTransaction`?
     // Check out Launchpad repo to view transaction states
     this.isWaiting = true
-    try {
-      await waitForTransaction(
-        addRecipient(
-          recipientRegistryAddress,
-          recipient,
-          registryInfo.deposit,
-          signer
-        ),
-        (hash) => (this.txHash = hash)
-      )
-    } catch (error) {
+    if (
+      recipientRegistryAddress &&
+      recipient &&
+      recipientRegistryInfo &&
+      currentUser
+    ) {
+      try {
+        await waitForTransaction(
+          addRecipient(
+            recipientRegistryAddress,
+            recipient,
+            recipientRegistryInfo.deposit,
+            currentUser.walletProvider.getSigner()
+          ),
+          (hash) => (this.txHash = hash)
+        )
+      } catch (error) {
+        this.isWaiting = false
+        this.txError = error.message
+        return
+      }
       this.isWaiting = false
-      this.txError = error.message
-      return
+
+      this.$router.push({
+        name: 'projectAdded',
```

Comment:
I've changed the router names. This should be `'project-added'`

Reply:
Good catch, my mistake on merging in prettier stuff.